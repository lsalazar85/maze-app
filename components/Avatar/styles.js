import styled from "styled-components";

export const AvatarContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50px;
  height: 50px;
  position: absolute;
  z-index: 10;
  left: ${({ x }) => x * 3.125 +'rem'};
  top: ${({ y }) => y * 3.125 +'rem'};
`