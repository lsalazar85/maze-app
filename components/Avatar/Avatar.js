import { AvatarContainer, AvatarIcon } from "./styles";

const Avatar = ({ x, y }) => {
    return(
        <AvatarContainer x={x} y={y}>
            <AvatarIcon />
        </AvatarContainer>
    )
}

export default Avatar