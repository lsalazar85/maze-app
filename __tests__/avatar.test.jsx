/**
 * @jest-environment jsdom
 */

import  { render } from '@testing-library/react'
import Avatar from "@/components/Avatar";

describe('Avatar', () => {
    const state = { x: 1, y: 0 }

    it('should render correctly', () => {
        const { asFragment } = render(<Avatar position={state} />);

        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    })
})