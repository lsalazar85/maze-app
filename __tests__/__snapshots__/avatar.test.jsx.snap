// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="styles__AvatarContainer-sc-djljgk-0 fhuxRF"
    x="1"
    y="0"
  >
    <div
      class="styles__AvatarIcon-sc-djljgk-1 lhygVp"
    />
  </div>
</DocumentFragment>
`;
